geocruncher/ComputeIntersections.py

```diff
@@ -1,93 +1,181 @@
 import math
 
+from typing import Tuple
 import numpy as np
 import pyvista as pv
-
+from gmlib.GeologicalModel3D import GeologicalModel
+from gmlib.architecture import from_GeoModeller, make_evaluator
 
 from .profiler.profiler import get_current_profiler
 from .off import read_off
 
-
 class MapSlice:
 
-    def output(xCoord, yCoord, nPoints, ranks, evaluate_z, isBase):
-        def computeRank(a, b):
-            return ranks([a, b, evaluate_z([a, b])]) - 1 if isBase else ranks([a, b, evaluate_z([a, b])])
+    @staticmethod
+    def output(xyz_reordered: np.array, n_points: int, model: GeologicalModel
+    ) -> list:
+        """Compute formation ranks for a top-down geological cross section.
+
+        Parameters
+        ----------
+        xyz_reordered: np.array
+            A numpy array of reordered (column by column -> x changes first, then y) coordinates, shape (n_points^2, 3).
+        n_points: int
+            Number of points in each dimension of the cross section.
+        model: gmlib.GeologicalModel3D.GeologicalModel
+            The GeologicalModel from gmlib to use for the formation rank evaluation.
 
-        def computeRankMatrix(index):
-            return np.array(list(map(computeRank, x[index], y[index]))).transpose().tolist()
+        Returns:
+            list: A list of ranks after evaluation, reshaped to (n_points, n_points).
+        
+        Notes
+        -----
+        The rank evaluation is being done without the topography, as it's not needed for a top-down view.
+        """
 
-        # x,y,z Coordinates expressed in real coordinates
-        xMapRange = np.linspace(xCoord[0], xCoord[1], nPoints)
-        yMapRange = np.linspace(yCoord[0], yCoord[1], nPoints)
+        cppmodel = from_GeoModeller(model)
+        evaluator = make_evaluator(cppmodel)
 
-        # x, z = np.ogrid[xCoord[0]:xCoord[1]:nPoints , zCoord[0]:zCoord[1]:nPoints]
-        y, x = np.meshgrid(yMapRange, xMapRange)
-        get_current_profiler().profile('map_grid')
```

Comment:
I moved the grid construction to the computations.py file because we need basically the same grid for both the rank and the faults (except of the reordering problem). Now we only have to compute the z-Values one time which is saves quite some time. But I moved also the profiler to there.